main.go

```diff
@@ -112,66 +121,127 @@ func main() {
+		if f.Name == flagName {
+			configSwitchIsSet = true
+		}
+	})
+	if !configSwitchIsSet {
+		if envConfigPath, ok := os.LookupEnv(envVarName); ok {
+			return envConfigPath
+		}
+	}
+	return ""
+}
+
+func initSentryClient(config *configuration.Registry) func() {
+	haltSentry, err := sentry.InitializeSentryClient(
+		ptr.String(config.GetSentryDSN()),
+		sentry.WithRelease(app.Commit),
+		sentry.WithEnvironment(config.GetEnvironment()),
+	)
+	if err != nil {
+		log.Panic(nil, map[string]interface{}{"err": err},
+			"failed to setup the sentry client")
+	}
+	return haltSentry
 }
 
 func printUserInfo() {
 	u, err := user.Current()
 	if err != nil {
-		log.Warn(nil, map[string]interface{}{
-			"err": err,
-		}, "failed to get current user")
+		log.Warn(nil, map[string]interface{}{"err": err},
+			"failed to get current user")
 	} else {
-		log.Info(nil, map[string]interface{}{
-			"username": u.Username,
-			"uuid":     u.Uid,
-		}, "Running as user name '%s' with UID %s.", u.Username, u.Uid)
+		log.Info(nil, map[string]interface{}{"username": u.Username, "uuid": u.Uid},
+			"Running as user name '%s' with UID %s.", u.Username, u.Uid)
 		g, err := user.LookupGroupId(u.Gid)
 		if err != nil {
-			log.Warn(nil, map[string]interface{}{
-				"err": err,
-			}, "failed to lookup group")
+			log.Warn(nil, map[string]interface{}{"err": err},
+				"failed to lookup group")
+		} else {
+			log.Info(nil, map[string]interface{}{"groupname": g.Name, "gid": g.Gid},
+				"Running as as group '%s' with GID %s.", g.Name, g.Gid)
+		}
+	}
+
+}
+
+func openDB(config *configuration.Registry) *gorm.DB {
+	for {
+		db, err := gorm.Open("postgres", config.GetPostgresConfigString())
+		if err != nil {
+			log.Logger().Errorf("ERROR: Unable to open connection to database %v", err)
+			log.Logger().Infof("Retrying to connect in %v...", config.GetPostgresConnectionRetrySleep())
+			time.Sleep(config.GetPostgresConnectionRetrySleep())
 		} else {
-			log.Info(nil, map[string]interface{}{
-				"groupname": g.Name,
-				"gid":       g.Gid,
-			}, "Running as as group '%s' with GID %s.", g.Name, g.Gid)
+			return db
 		}
 	}
+}
 
+func setupDB(db *gorm.DB, config *configuration.Registry) {
+	if config.IsPostgresDeveloperModeEnabled() && log.IsDebug() {
+		db = db.Debug()
+		log.Logger().Infof("Started DB debug mode")
+	}
+	if config.GetPostgresConnectionMaxIdle() > 0 {
+		db.DB().SetMaxIdleConns(config.GetPostgresConnectionMaxIdle())
+		log.Logger().Infof("Configured connection pool max idle %v", config.GetPostgresConnectionMaxIdle())
+	}
+	if config.GetPostgresConnectionMaxOpen() > 0 {
+		db.DB().SetMaxOpenConns(config.GetPostgresConnectionMaxOpen())
+		log.Logger().Infof("Configured connection pool max open %v", config.GetPostgresConnectionMaxOpen())
+	}
+	application.SetDatabaseTransactionTimeout(config.GetPostgresTransactionTimeout())
+}
+
+func getTokenManager(config *configuration.Registry) token.Manager {
+	tokenMgr, err := token.NewManager(config)
```

Comment:
I would recommend to use the default token manager instead - https://github.com/fabric8-services/fabric8-auth/blob/master/main.go#L151
The default manager is available in the latest common.